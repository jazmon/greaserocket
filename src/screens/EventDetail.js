// @flow
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import theme from 'constants/theme';

type Props = {
  navigation: Object,
  route: Object,
  loading: boolean,
};

type State = {
  text: string,
};

class EventDetail extends React.Component {
  props: Props;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: 'EventDetail',
    };
  }

  state: State;

  render() {
    const { text } = this.state;
    return (
      <View style={styles.container}>
        <Text style={styles.text}>{text}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: theme.WHITE,
  },
  text: {
    color: theme.BLACK,
    fontSize: 16,
  },
});

export const EventDetailComponent = EventDetail;
export default EventDetail;
