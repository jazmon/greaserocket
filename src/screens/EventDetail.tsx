// @flow
import * as React from 'react';
import styled from 'styled-components/native';
import { ThemeType } from 'constants/theme';

type Props = {
  navigation: Object;
  route: Object;
  loading: boolean;
};

type State = {
  text: string;
};

const Container = styled.View`
  flex: 1;
  flex-direction: column;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
`;
const BaseText = styled.Text`
  color: ${({ theme }: { theme: ThemeType }) => theme.text.colors.primary};
  font-size: 16px;
`;

class EventDetail extends React.Component<Props, State> {
  props: Props;

  constructor(props: Props) {
    super(props);

    this.state = {
      text: 'EventDetail',
    };
  }

  state: State;

  render() {
    const { text } = this.state;
    return (
      <Container>
        <BaseText>
          {text}
        </BaseText>
      </Container>
    );
  }
}

export const EventDetailComponent = EventDetail;
export default EventDetail;
