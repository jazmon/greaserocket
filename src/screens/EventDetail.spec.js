import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import { EventDetailComponent as EventDetail } from './EventDetail';


it('renders correctly', () => {
  const tree = renderer.create(
    <EventDetail />
  );
  expect(tree).toMatchSnapshot();
});
