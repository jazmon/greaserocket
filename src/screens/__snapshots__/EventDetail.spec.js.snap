exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#000",
        "fontSize": 16,
      }
    }>
    EventDetail
  </Text>
</View>
`;
